kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/TopicsService.java

```diff
@@ -180,6 +187,25 @@
         .map(clusterMapper::toTopic);
   }
 
+  public Mono<TopicDTO> recreateTopic(KafkaCluster cluster, String topicName) {
+    return adminClientService.get(cluster)
+            .flatMap(cl -> loadTopic(cluster, topicName))
+            .flatMap(topic -> deleteTopic(cluster, topicName).thenReturn(new TopicCreationDTO()
+                            .name(topic.getName())
+                            .partitions(topic.getPartitionCount())
+                            .replicationFactor(topic.getReplicationFactor())
+                            .configs(topic
+                                    .getTopicConfigs()
+                                    .stream()
+                                    .collect(Collectors.toMap(InternalTopicConfig::getName,
+                                            InternalTopicConfig::getValue))))
+                    .delayElement(Duration.ofSeconds(recreateTopicTimeout))
+                    .flatMap(topicCreation -> createTopic(cluster,  Mono.just(topicCreation))
+                            .retryWhen(Retry.fixedDelay(maxAttemptToRecreateTopic,
+                                            Duration.ofSeconds(recreateTopicTimeout))
+                                    .filter(throwable -> throwable instanceof TopicMetadataException))));
```

Comment:
1. 
```
            Retry.fixedDelay(recreateDelayMaxAttempts, Duration.ofSeconds(recreateDelayInSeconds))
                .filter(throwable -> filter here...)
                .onRetryExhaustedThrow(
                    new TopicCreationException("Can't created topic '%s' in %d seconds: " +
                        "topic deletion is still in progress", topicName, (recreateDelayMaxAttempts*recreateDelayInSeconds))
                )
```

2. timeout naming is not very good IMHO - I think timeout means whole time taken for operation, including retry. So, I think we should rename.

3. I think we should do retry more granularly - I would vote for maxAttemptToRecreateTopic=15, recreateDelayInSeconds=1

4. Current private createTopic wraps all kafka exceptions with our TopicMetadataException  - it is not good to use this exception for retry since it can mean anything, not just "topic already exists" -> add new createTopic method without exceptions/use adminclient and filter kafka exception

Reply:
Fixed.